scripts/setup_sgtm_tasks_project.py

```diff
@@ -0,0 +1,228 @@
+            EnumOption(name="Success", color="green"),
+            EnumOption(name="Failure", color="red"),
+        ],
+    ),
+]
+
+
+def parse_args():
+    parser = argparse.ArgumentParser(
+        description="""A helper script to set up an SGTM tasks Asana project with the suggested custom fields
+		"""
+    )
+
+    parser.add_argument(
+        "-p",
+        "--personal_access_token",
+        help="""The Asana Personal Access Token that will be used to set up the Asana project. If you do not already have one,
+		follow the instructions at https://developers.asana.com/docs/personal-access-token
+		""",
+        type=str,
+        required=True,
+    )
+
+    subparsers = parser.add_subparsers(
+        title="Sub-commands",
+    )
+
+    parser_create = subparsers.add_parser(
+        "create",
+        help="""Use this option if you want to set up a new project from scratch""",
+    )
+
+    parser_create.add_argument(
+        "-n",
+        "--new_project_name",
+        help="""The Asana Project name""",
+        type=str,
+        required=True,
+    )
+
+    parser_create.add_argument(
+        "-t",
+        "--team_id",
+        help="""This is the global ID of the team that you want to share the project with. You can find this in the URL
+        by navigating to the desired team in Asana
+		""",
+        type=str,
+        required=True,
+    )
+
+    parser_create.set_defaults(func=setup_new_project)
+
+    parser_update = subparsers.add_parser(
+        "update",
+        help="""Use this option if you have an existing Asana Project that you want to use to sync 
+        SGTM tasks""",
+    )
+
+    parser_update.add_argument(
+        "-e",
+        "--existing_project_id",
+        help="""The global ID of an existing Asana Project""",
+        type=str,
+        required=True,
+    )
+
+    parser_update.set_defaults(func=setup_existing_project)
+
+    return parser.parse_args()
+
+
+def setup_new_project(args) -> None:
+    client = AsanaClient(args.personal_access_token)
+    project_id = client.create_project(args.new_project_name, args.team_id)
+    client.setup_custom_fields(project_id)
+
+
+def setup_existing_project(args) -> None:
+    client = AsanaClient(args.personal_access_token)
+    client.setup_custom_fields(args.existing_project_id)
+    client.add_user_to_project(args.existing_project_id)
+
+
+class AsanaClient(object):
+    def __init__(self, personal_access_token: str):
+        self.client = asana.Client.access_token(personal_access_token)
+        self.client.headers = {"Asana-Enable": "string_ids"}
+        self.workspace_id = self._get_workspace_id()
+
+    def _get_workspace_id(self) -> str:
+        """
+        Get the workspace ID that the user who created the PAT is a part of
+        """
+        pat_user = self.client.users.me()
+        if len(pat_user["workspaces"]) == 1:
+            return pat_user["workspaces"][0]["gid"]
+        else:
+            workspaces = pat_user["workspaces"]
```

Comment:
Tested this manually in the python console by setting up some fake data - I wasn't sure how else to test this